pkg/roachprod/vm/azure/azure.go

```diff
@@ -317,37 +353,33 @@ func (p *Provider) Create(
 	ctx, cancel := context.WithTimeout(context.Background(), p.OperationTimeout)
 	defer cancel()
 
-	if len(providerOpts.Locations) == 0 {
-		if opts.GeoDistributed {
-			providerOpts.Locations = DefaultLocations
-		} else {
-			providerOpts.Locations = []string{DefaultLocations[0]}
-		}
+	zones, err := parseZones(opts, providerOpts)
+	if err != nil {
+		return err
 	}
 
-	if len(providerOpts.Zone) == 0 {
-		providerOpts.Zone = defaultZone
+	// Effectively a map of node number to zone.
+	nodeZones := vm.ZonePlacement(len(zones), len(names))
+	// Invert it.
+	zoneToHostNames := make(map[Zone][]string, min(len(zones), len(names)))
+	for i, name := range names {
+		zone := zones[nodeZones[i]]
+		zoneToHostNames[zone] = append(zoneToHostNames[zone], name)
 	}
 
-	if _, err := p.createVNets(l, ctx, providerOpts.Locations, *providerOpts); err != nil {
-		return err
+	uniqueLocations := make(map[string]struct{})
+	for zone := range zoneToHostNames {
+		uniqueLocations[zone.Location] = struct{}{}
 	}
 
-	// Effectively a map of node number to location.
-	nodeLocations := vm.ZonePlacement(len(providerOpts.Locations), len(names))
-	// Invert it.
-	nodesByLocIdx := make(map[int][]int, len(providerOpts.Locations))
-	for nodeIdx, locIdx := range nodeLocations {
-		nodesByLocIdx[locIdx] = append(nodesByLocIdx[locIdx], nodeIdx)
+	if _, err := p.createVNets(l, ctx, maps.Keys(uniqueLocations), *providerOpts); err != nil {
+		return err
 	}
 
 	errs, _ := errgroup.WithContext(ctx)
-	for locIdx, nodes := range nodesByLocIdx {
-		// Shadow variables for closure.
-		locIdx := locIdx
-		nodes := nodes
```

Comment:
For consistency, it would be nice to reuse the same `printf` as in gcloud.go,

```
l.Printf("Creating %d instances, distributed across [%s]", len(names), strings.Join(usedZones, ", "))
```

where `usedZones` is the set of zones.

Reply:
Done.